File: Data Structures & Algorithms/two-integer-sum/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {

        let seen = [];

        for(let i = 0; i < nums.length; i++)
        {
            let current = nums[i];
            let lookingFor = target - current;
            if(seen.includes(lookingFor)) {return [seen.indexOf(lookingFor),i];}
            
            if(!seen.includes(current)) {seen.push(current);}
        }
        return [0,0];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.Length - 1;

        while (left < right)
        { 
            int total = numbers[left] + numbers[right];
            if(total == target) {return [left + 1, right + 1];}

            if(total < target) {left++;}
            else {right--;}
        }

        return [0,0];
    }
}
